src/main/drivers/accgyro/accgyro_spi_lsm6dso.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#ifdef USE_ACCGYRO_LSM6DSO
+
+#include "drivers/accgyro/accgyro.h"
+#include "drivers/accgyro/accgyro_spi_lsm6dso.h"
+#include "drivers/bus_spi.h"
+#include "drivers/exti.h"
+#include "drivers/io.h"
+#include "drivers/io_impl.h"
+
+#if defined(USE_GYRO_EXTI) && defined(USE_MPU_DATA_READY_SIGNAL)
+void lsm6dsoExtiHandler(extiCallbackRec_t *cb)
+{
+    gyroDev_t *gyro = container_of(cb, gyroDev_t, exti);
+    gyro->dataReady = true;
+}
+#endif
+
+bool lsm6dsoAccRead(accDev_t *acc)
+{
+    enum {
+        IDX_REG = 0,
+        IDX_ACCEL_XOUT_L,
+        IDX_ACCEL_XOUT_H,
+        IDX_ACCEL_YOUT_L,
+        IDX_ACCEL_YOUT_H,
+        IDX_ACCEL_ZOUT_L,
+        IDX_ACCEL_ZOUT_H,
+        BUFFER_SIZE,
+    };
+
+    uint8_t lsm6dso_rx_buf[BUFFER_SIZE];
+    static const uint8_t lsm6dso_tx_buf[BUFFER_SIZE] = {LSM6DSO_REG_OUTX_L_A | 0x80, 0, 0, 0, 0, 0, 0};
+
+    IOLo(acc->bus.busdev_u.spi.csnPin);
+    spiTransfer(acc->bus.busdev_u.spi.instance, lsm6dso_tx_buf, lsm6dso_rx_buf, BUFFER_SIZE);   // receive response
```

Comment:
This should be using `busReadRegisterBuffer()` to be transaction aware. Or we should be pushing all of the SPI transaction support down into `src/main/drivers/bus_spi.c`.

Reply:
Updated to support SPI transactions.